xrpl/transaction/oracle_delete.go

```diff
@@ -0,0 +1,38 @@
+package transaction
+
+// Deletes a price oracle for the Account.
+// ```json
+//
+//	{
+//	  "TransactionType": "OracleDelete",
+//	  "Account": "rsA2LpzuawewSBQXkiju3YQTMzW13pAAdW",
+//	  "OracleDocumentID": 34
+//	}
+//
+// ```
+type OracleDelete struct {
+	BaseTx
+	// A unique identifier of the price oracle for the Account.
+	OracleDocumentID uint32
+}
+
+// Returns the type of the transaction.
+func (tx *OracleDelete) TxType() TxType {
+	return OracleDeleteTx
+}
+
+// Returns a flattened transaction.
+func (tx *OracleDelete) Flatten() FlatTransaction {
+	flattened := tx.BaseTx.Flatten()
+
+	flattened["TransactionType"] = tx.TxType().String()
+
+	flattened["OracleDocumentID"] = tx.OracleDocumentID
+
+	return flattened
+}
+
+// Validates the transaction.
+func (tx *OracleDelete) Validate() (bool, error) {
```

Comment:
Shouldn't we check that `OracleDocumentID` is set? It is a required field.

Reply:
It will always have a value since integers in go are always initialized to zero value. I didn't find any restrictions about the `OracleDocumentID` range value, so I imagine that it can have any value within the `uint32` value range.